addons/mail/controllers/common.py

```diff
@@ -0,0 +1,55 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.http import request
+
+
+class MailControllerCommon(http.Controller):
+    """A base controller class that can serve as a foundation for mail controllers
+    and provides shared methods."""
+
+    # access helpers
+    # ------------------------------------------------------------
+
+    @classmethod
+    def _get_message_with_access(cls, message_id, mode="read", **kwargs):
```

Comment:
As `cls` is never used, does overriding still works if this uses `@staticmethod` instead?

Reply:
Yes, it also works well as a static method.